chore: move apps and tools to their own directory

LNSD opened the pull request:
This is another gardening 🪴 PR:

- Moved apps to their directory under `/apps` directory
- Moved tools to their directory under `/tools` directory
- Updated tests accordingly
- Updated build scripts accordingly

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6380a439 | [#1](https://ci.status.im/job/nim-waku/job/prs/job/linux/job/PR-1228/1/) | 2022-10-05 10:53:57 | ~15 min | `linux` | [:package:`bin`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/wakunode2-221005-103756-6380a4-linux.bin) |

jm-clius reviewed (approved):
Much better than what we had! @alrevuelta, in this scheme I'd imagine the canary service would go under `tools`.

LNSD commented:
> Much better than what we had! @alrevuelta, in this scheme I'd imagine the canary service would go under `tools`.

Indeed. @alrevuelta put already the _wakucanary_ tool under `/tools` dir in #1205

alrevuelta reviewed (commented):
Great!

Some questions:
* is `waku/v2/node/scripts/rpc_query` binary a leftover?
* will reuse this for #1205. mind elaborating the difference between `scripts` and `simulation`?
* should we add a readme with a howto for each tool?

LNSD commented on waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim:
```diff
@@ -17,7 +17,7 @@ import
   ../../utils/time,
   ../../node/[wakunode2_types,config],
-  ../../../../../examples/v2/config_chat2,
+  ../../../../../apps/chat2/config_chat2,
```
I know, I know...

alrevuelta replied:
Is this something we aim to fix? Already tracked in an issue? :)

LNSD replied:
The issue: https://github.com/status-im/nwaku/issues/1222

alrevuelta replied:
Great, solving and moving the conversations to the other one.

LNSD commented on apps/chat2/chat2.nim:
```diff
@@ -21,16 +21,17 @@ import libp2p/[switch,                   # manage transports, a single entry poi
                protocols/secure/secio,   # define the protocol of secure input / output, allows encrypted communication that uses public keys to validate signed messages instead of a certificate authority like in TLS
```
Some context/background: _I want to decouple the `wkaunode2` binary from the libraries._

This is the best name I came up with that covers the production binaries/fully fledged apps like `wakunode2` or the `wakubridge`. I am open to suggestions. BTW I discarded the following for the same reason: `binaries`, `bin`, and `cmd`.

Regarding `chat2`... Well, I don't know how to categorize it. But maybe we can use it to show how things should not be done 😆

alrevuelta replied:
Okay, so as I see this comment is related to my other comment where I ask about prod code depending on `chat2` "example". What do you think if we:
* fix the `chat2` [issue](https://github.com/status-im/nwaku/pull/1228#discussion_r987802325), moving the production code to somewhere in `waku`.
* then in `chat2` import that code, just like the rest of the `waku` imports.
* leave `examples` folder as it it, since i think its very important to keep that naming. i.e. [nim-libp2p](https://github.com/status-im/nim-libp2p/blob/unstable/examples/) also has it.

with this we are addressing your concern of "I want to decouple the wkaunode2 binary from the libraries." right?

LNSD replied:
It is not easy fix, since it implies RLN module code changes. The circular dependency issue is tracked here: https://github.com/status-im/nwaku/issues/1222

We should tackle that in a separate work package. This is one of the reasons why I was asking for some bandwidth to shape the waku RLN module code in this morning meeting 😄

LNSD replied:
> with this we are addressing your concern of "I want to decouple the wkaunode2 binary from the libraries." right?

No... That's a bigger work package. Currently, the `wakunode2.nim` file has two parts:

* The node main routine (that does the setup of the storage, and spins up the different protocols and many other things).
* The waku node type definitions and associated procedures.

This "unit" is too big and does too many things. From a maintenance and debugging point of view, it has been proven that we need to simplify it. How? That's another, longer, discussion.

LNSD commented:
> * is `waku/v2/node/scripts/rpc_query` binary a leftover?

These are scripts that were used at some point, but I am not aware if they have been used in the last 4-5 months. And added to that, we are replacing the JSON-RPC with a REST API, so... I am not sure about the future of those tools. In this PR, I have just moved them out of the `waku/v2/node` directory since are utilities/tools, as per my understanding.

> will reuse this for https://github.com/status-im/nwaku/pull/1205. mind elaborating the difference between scripts and simulation?

By reading the code, the `scritps` are just utility tools to interact with a nwaku node through the JSON-RPC API. The `simulations` are other scripts that were used in the past to do some relay network simulations. And the same thing as with the `scripts`, I have not seen them being used in the last 4-5 months.

> should we add a readme with a howto for each tool?

I agree with that. That would be very helpful. 

But unfortunately, I don't know much about these binaries 😅 And, at some point, I think we'll remove them from the repo since they are unmaintained and unused. So I am unsure if there's any value in adding a readme right now. Wdyt?

alrevuelta commented:
>>  is waku/v2/node/scripts/rpc_query binary a leftover?

> These are scripts that were used at some point, but I am not aware if they have been used in the last 4-5 months. And added to that, we are replacing the JSON-RPC with a REST API, so... I am not sure about the future of those tools. In this PR, I have just moved them out of the waku/v2/node directory since are utilities/tools, as per my understanding.

Was asking about the binary, not the scripts. Its unusual to see binaries in a repo, so that's why I was asking if it was a leftover that someone left there, and you were just removing it. Or if there were any other reason behind.

> But unfortunately, I don't know much about these binaries 😅 And, at some point, I think we'll remove them from the repo since they are unmaintained and unused. So I am unsure if there's any value in adding a readme right now. Wdyt?

Ah okay, I see.

LNSD commented:
> Was asking about the binary, not the scripts. Its unusual to see binaries in a repo, so that's why I was asking if it was a leftover that someone left there, and you were just removing it. Or if there were any other reason behind.

AH! Sorry, my bad. Yes... That happens when someone commits a binary artifact with the sources. This is not the first binary that was committed and I remove 😁

alrevuelta commented:
> AH! Sorry, my bad. Yes... That happens when someone commits a binary artifact with the sources. This is not the first binary that was committed and I remove 😁

I see. Perahps beyond the PR but maybe we can add a rule to ignore files without extensions. [example](https://stackoverflow.com/questions/19023550/how-do-i-add-files-without-dots-in-them-all-extension-less-files-to-the-gitign)
